### PR TITLE
refactor: Feature Flag structure refactoring

### PR DESCRIPTION
**Pull request by 0x656b694d**
### Description

Refactoring.

### User-facing documentation

(*must be* 2 items and both *must be* checked)

- [x] CHANGELOG update is not needed
- [x] Documentation is not needed

### Testing

- [x] inspected CI results

#### Automated testing

(*must be* at least 1 item and all items *must be* checked)

- [x] added unit tests
- [x] modified existing tests

#### How I validated my change

Unit tests.

### CODE REVIEW

File: pkg/features/feature.go [hunk 1]
```diff
@@ -3,21 +3,36 @@ package features
...
+	defaultValueBit
+	unchangeableBit
+)
+
+const (
+	// Const integer 1 for release, 0 for development build:
+	releaseBuild int = 1 - (len(buildinfo.BuildFlavor)-len("release"))/4
+
+	// mode values:
+	devPreview         mode = 0 * releaseStageBit
```
Comment (stehessel) [inline pkg/features/feature.go:24]: Why not use the option pattern here? I think that would be less complicated and works better when modes are orthogonal like `devPreview`/`techPreview` and `enabled`/`disabled`.

Comment (0x656b694d) [reply pkg/features/feature.go:24]: Dunno, for me the bit mask works just fine for orthogonal boolean values:

```go
ScannerV4Support = registerFeature(enabled|unchangeableInProd, "Enable Scanner V4 Support", "ROX_SCANNER_V4_SUPPORT")
// vs
ScannerV4Support = registerFeature("Enable Scanner V4 Support", "ROX_SCANNER_V4_SUPPORT", withEnabled(true), withUnchangeableInProd(true))
```

Comment (0x656b694d) [reply pkg/features/feature.go]: See #12101 with the Options implementation.

**Comment by 0x656b694d** [comment]
Closed in favor of #12101.

**Pull request closed without merging**